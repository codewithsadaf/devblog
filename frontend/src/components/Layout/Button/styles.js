//imports
import styled, { css } from "styled-components";

const handleButtonVariant = {
    default: (theme) => css`
        background: ${theme.colors.primaryColor};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: transparent;
            color: ${theme.colors.primaryColor};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,
    outline: (theme) => css`
        background: transparent;
        color: ${theme.colors.primaryColor};
        border: 2px solid ${theme.colors.primaryColor};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            background: ${theme.colors.primaryColor};
            color: ${theme.colors.white};
            border: 2px solid ${theme.colors.primaryColor};
            transition: ${theme.transition.tran_03};
        }
    `,

    success: (theme) => css`
        background: ${theme.colors.success};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.success};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,

    danger: (theme) => css`
        background: ${theme.colors.danger};
        color: ${theme.colors.white};
        border: 2px solid ${theme.colors.danger};
        border-radius: 4px;
        transition: ${theme.transition.tran_03};

        &:hover {
            opacity: 0.8;
            transition: ${theme.transition.tran_03};
        }
    `,
};

const handleUppercase = () => css`
    text-transform: uppercase;
`;

const handleMaxWidth = () => css`
    width: 100%;
    padding: 1em 0;
`;

const handleSmallStyles = (theme) => css`
    padding: 7px 30px;
    border-radius: 4px;
    font-size: ${theme.font.sizes.small};
    border: 1px solid;

    &:hover {
        border: 1px solid;
    }
`;

export const ButtonContainer = styled.button`
    ${({ theme, buttonVariant, uppercase, maxWidth, isSmall }) => css`
        padding: 0.5em 1.8em;
        font-size: 1em;
        border: none;
        font-weight: bold;
        cursor: pointer;
        ${handleButtonVariant[buttonVariant](theme)};
        ${uppercase && handleUppercase()};
        ${maxWidth && handleMaxWidth()};
        ${isSmall && handleSmallStyles(theme)};
    `}
`;
