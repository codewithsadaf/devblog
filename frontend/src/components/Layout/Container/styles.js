//imports
import styled from "styled-components";

export const Container = styled.div`
    min-height: 60vh;
    margin-bottom: 5em;
    padding: 0 2em;
`;
