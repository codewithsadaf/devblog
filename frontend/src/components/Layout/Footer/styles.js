//imports
import styled, { css } from "styled-components";

export const FooterWrapper = styled.footer`
    ${({ theme }) => css`
        height: 250px;
        width: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        background-color: ${theme.colors.footerBackground};
        text-align: center;
    `}
`;
export const FooterHeading = styled.h3`
    ${({ theme }) => css`
        @media ${theme.media.lteMedium} {
            font-size: ${theme.font.sizes.medium};
            line-height: ${theme.font.line_height.h6};
        }
    `}
`;

export const CopyrightText = styled.span`
    display: block;
    margin-top: 0.8em;
`;
