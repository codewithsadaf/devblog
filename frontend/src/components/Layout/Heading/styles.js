//imports
import styled, { css } from "styled-components";

const headingSize = {
    huge: (theme) => css`
        font-size: ${theme.font.sizes.xhuge};
        line-height: ${theme.font.line_height.h1};
        ${mediaFont(theme)};
    `,
    big: (theme) => css`
        font-size: ${theme.font.sizes.huge};
        line-height: ${theme.font.line_height.h2};
    `,
    large: (theme) => css`
        font-size: ${theme.font.sizes.xxlarge};
        line-height: ${theme.font.line_height.h3};
    `,
    medium: (theme) => css`
        font-size: ${theme.font.sizes.medium};
        line-height: ${theme.font.line_height.h6};
    `,
    small: (theme) => css`
        font-size: ${theme.font.sizes.small};
        line-height: ${theme.font.line_height.small};
    `,
    xsmall: (theme) => css`
        font-size: ${theme.font.sizes.xsmall};
        line-height: ${theme.font.line_height.small};
    `,
};

const mediaFont = (theme) => css`
    @media ${theme.media.lteMedium} {
        font-size: ${theme.font.sizes.xlarge};
        line-height: ${theme.font.line_height.h4};
    }
`;

export const Container = styled.h1`
    ${({ theme, size, uppercase, bold }) => css`
        text-transform: ${uppercase ? "uppercase" : "none"};
        font-weight: ${bold ? "bold" : 500};
        ${headingSize[size](theme)};
    `}
`;
