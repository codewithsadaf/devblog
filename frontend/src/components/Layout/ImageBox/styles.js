//imports
import styled, { css } from "styled-components";

const handleBigStyle = () => css`
    width: 100%;
    height: auto;
`;

export const Container = styled.div`
    ${({ theme }) => css``}
`;

export const ImageElm = styled.img`
    ${({ rounded }) => css`
        width: 150px;
        height: 150px;
        object-fit: cover;
        display: block;
        margin: 1em auto;
        border-radius: ${rounded ? "50%" : "7px"};
    `}
`;
