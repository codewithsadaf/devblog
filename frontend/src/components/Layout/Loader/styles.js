//imports
import styled, { css } from "styled-components";

export const Container = styled.div`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100vh;
    background: rgba(0, 0, 0, 0.6);
    z-index: 99999;
    pointer-events: all;
`;

export const Circle = styled.div`
    ${({ theme }) => css`
        &,
        &::before,
        &::after {
            border-radius: 50%;
            width: 2.5em;
            height: 2.5em;
            animation-fill-mode: both;
            animation: bblFadInOut 1.8s infinite ease-in-out;
        }

        & {
            color: ${theme.colors.white};
            font-size: 7px;
            position: relative;
            text-indent: -9999em;
            transform: translateZ(0);
            animation-delay: -0.16s;
        }

        &::before,
        &:after {
            content: "";
            position: absolute;
            top: 0;
        }

        &::before {
            left: -3.5em;
            animation-delay: -0.32s;
        }

        &::after {
            left: 3.5em;
        }

        @keyframes bblFadInOut {
            0%,
            80%,
            100% {
                box-shadow: 0 2.5em 0 -1.3em;
            }
            40% {
                box-shadow: 0 2.5em 0 0;
            }
        }
    `}
`;
