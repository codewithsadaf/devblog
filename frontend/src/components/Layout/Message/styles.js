//imports
import styled, { css } from "styled-components";
//react icons
import { IoClose } from "react-icons/io5";

export const Container = styled.div`
    ${({ theme }) => css`
        display: flex;
        align-items: center;
        justify-content: space-between;
        position: fixed;
        bottom: 30px;
        left: 50%;
        transform: translateX(-50%);
        padding: 25px 30px;
        min-width: 25%;
        border-radius: 7px;
        z-index: 9999;
        font-weight: bold;
        box-shadow: rgba(0, 0, 0, 0.24) 0px 3px 8px;
        transition: ${theme.transition.tran_03};

        &.success {
            background-color: ${theme.colors.success};
            color: ${theme.colors.white};
        }

        &.error {
            background-color: ${theme.colors.danger};
            color: ${theme.colors.white};
        }

        @media ${theme.media.lteMedium} {
            bottom: 0;
            right: 50%;
            left: 50%;
            min-width: 90%;
            transform: translate(-50%, -50%);
        }
    `}
`;

export const CloseIcon = styled(IoClose)`
    ${({ theme }) => css`
        color: ${theme.colors.white};
        cursor: pointer;
        font-size: 19px;
        margin-left: 0.3em;
    `}
`;

export const MessageText = styled.span`
    display: block;
`;
