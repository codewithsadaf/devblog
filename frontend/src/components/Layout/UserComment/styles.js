//imports
import styled, { css } from "styled-components";
//components
import { Text as TextComponent } from "../../../components/Layout/TextComponent/styles";
import { ImageElm } from "../ImageBox/styles";
import { InputComponent } from "../../Form/Input/styles";
//react icons
import { IoCheckmark } from "react-icons/io5";
import { MdEdit, MdClose } from "react-icons/md";
import { FaTrash } from "react-icons/fa";

export const Container = styled.div`
    ${({ theme }) => css`
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: flex-start;
        margin: 2em 0;
        padding: 0 7px;
        position: relative;
    `}
`;

export const ProfileImageContainer = styled.div`
    ${({ theme }) => css`
        min-width: 7%;
        width: 7%;
        height: 100%;
        margin: 0 1em 0 0;

        ${ImageElm} {
            width: 100%;
            height: 100%;
            margin: 0;
        }

        @media ${theme.media.lteMedium} {
            width: 10%;
            min-width: 10%;
        }
    `}
`;

export const CommentContainer = styled.div`
    ${({ theme }) => css`
        width: 93%;
        border-bottom: 1px solid ${theme.colors.gray};
        padding-bottom: 1em;

        & > ${TextComponent} {
            margin-top: 1.2em;
            display: block;
            width: 100%;
            word-wrap: break-word;
        }

        @media ${theme.media.lteMedium} {
            width: 90%;
        }
    `}
`;

export const ActionsWrapper = styled.div`
    position: absolute;
    right: 0;
    top: 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
`;

export const RemoveIcon = styled(FaTrash)`
    ${({ theme }) => css`
        cursor: pointer;
        transition: ${theme.transition.tran_03};
        font-size: ${theme.font.sizes.default};

        &:hover {
            color: ${theme.colors.danger};
            transition: ${theme.transition.tran_03};
        }
    `}
`;

export const EditIcon = styled(MdEdit)`
    ${({ theme }) => css`
        cursor: pointer;
        margin-right: 0.5em;
        transition: ${theme.transition.tran_03};
        font-size: ${theme.font.sizes.default};

        &:hover {
            color: ${theme.colors.success};
            transition: ${theme.transition.tran_03};
        }
    `}
`;

export const EditInput = styled(InputComponent)`
    border: none;
    width: 100%;
    margin-top: 1.2em;

    &:focus {
        border: none;
    }
`;

export const ProceedIcon = styled(IoCheckmark)`
    ${({ theme }) => css`
        cursor: pointer;
        margin-right: 0.5em;
        transition: ${theme.transition.tran_03};
        color: ${theme.colors.success};
        font-size: ${theme.font.sizes.medium};

        &:hover {
            transition: ${theme.transition.tran_03};
            color: ${theme.colors.primaryColor};
        }
    `}
`;

export const CloseIcon = styled(MdClose)`
    ${({ theme }) => css`
        cursor: pointer;
        transition: ${theme.transition.tran_03};
        color: ${theme.colors.danger};
        font-size: ${theme.font.sizes.medium};

        &:hover {
            transition: ${theme.transition.tran_03};
            color: ${theme.colors.primaryColor};
        }
    `}
`;
