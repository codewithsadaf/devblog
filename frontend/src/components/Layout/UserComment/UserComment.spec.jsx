//imports
import { fireEvent, screen } from "@testing-library/react";
//styles
import { renderTheme } from "../../../styles/render-theme";
import { theme } from "../../../styles/theme";
//components
import { UserComment } from "./";

let props = {
    userName: "John Doe",
    creationDate: "2023-04-29",
    commentText: "Lorem ipsum dolor sit amet",
    imageSrc: "https://example.com/user.jpg",
    showActions: false,
    removeCommentFn: jest.fn(),
    editCommentFn: jest.fn(),
    successEditFn: jest.fn(),
    commentId: "1",
    postId: "1",
};

describe("<UserComment />", () => {
    describe("Rendering", () => {
        it("should render UserComment", () => {
            renderTheme(<UserComment {...props} />);

            expect(screen.getByText(props.userName));
            expect(screen.getByText(props.creationDate));
            expect(screen.getByText(props.commentText));
            expect(screen.getByRole("img")).toHaveAttribute(
                "src",
                props.imageSrc
            );
        });

        it("should not render user actions of edit and remove when showActions is false", () => {
            renderTheme(<UserComment {...props} />);

            expect(
                screen.queryByLabelText("Edit Icon")
            ).not.toBeInTheDocument();
            expect(
                screen.queryByLabelText("Remove Icon")
            ).not.toBeInTheDocument();
        });

        it("should render user actions of edit and remove when showActions is true", () => {
            renderTheme(<UserComment {...props} showActions={true} />);

            expect(screen.getByLabelText("Edit Icon")).toBeInTheDocument();
            expect(screen.getByLabelText("Remove Icon")).toBeInTheDocument();
        });

        it("should increase ProfileImageContainer width on mobile", () => {
            renderTheme(<UserComment {...props} />);

            expect(
                screen.getByRole("img", { name: "User Image" }).parentElement
                    .parentElement
            ).toHaveStyleRule("width", "10%", {
                media: theme.media.lteMedium,
            });
        });

        it("should decrease CommentContainer width on mobile", () => {
            renderTheme(<UserComment {...props} />);

            expect(
                screen.getByText(props.userName).parentElement
            ).toHaveStyleRule("width", "90%", {
                media: theme.media.lteMedium,
            });
        });

        it("should match snapshot", () => {
            const { container } = renderTheme(<UserComment {...props} />);

            expect(container.firstChild).toMatchSnapshot();
        });
    });

    describe("Functionality", () => {
        beforeEach(() => {
            renderTheme(<UserComment {...props} showActions={true} />);
        });

        it("should render edit textarea when edit icon is clicked", () => {
            fireEvent.click(screen.getByLabelText("Edit Icon"));

            expect(
                screen.getByPlaceholderText("Edite seu comentário aqui..")
            ).toBeInTheDocument();
        });

        it("should call removeCommentFn when remove icon is clicked", () => {
            fireEvent.click(screen.getByLabelText("Remove Icon"));

            expect(props.removeCommentFn).toHaveBeenCalledTimes(1);
        });

        it("should close edit textarea on close icon click", () => {
            fireEvent.click(screen.getByLabelText("Edit Icon"));

            expect(
                screen.getByPlaceholderText("Edite seu comentário aqui..")
            ).toBeInTheDocument();

            fireEvent.click(screen.getByLabelText("Close Icon"));

            expect(
                screen.queryByPlaceholderText("Edite seu comentário aqui..")
            ).not.toBeInTheDocument();
        });

        it("should call edit function on proceed icon click and close edit textarea", () => {
            fireEvent.click(screen.getByLabelText("Edit Icon"));

            expect(
                screen.getByPlaceholderText("Edite seu comentário aqui..")
            ).toBeInTheDocument();

            fireEvent.click(screen.getByLabelText("Proceed Icon"));

            expect(props.editCommentFn).toHaveBeenCalledTimes(1);

            expect(
                screen.queryByPlaceholderText("Edite seu comentário aqui..")
            ).not.toBeInTheDocument();
        });

        it("should call successEditFn", () => {
            const newCommentText = "New Comment Text";

            jest.spyOn(props, "editCommentFn").mockImplementation(
                props.successEditFn
            );

            fireEvent.click(screen.getByLabelText("Edit Icon"));

            expect(
                screen.getByPlaceholderText("Edite seu comentário aqui..")
            ).toBeInTheDocument();

            fireEvent.change(
                screen.getByPlaceholderText("Edite seu comentário aqui.."),
                { target: { value: newCommentText } }
            );

            fireEvent.click(screen.getByLabelText("Proceed Icon"));

            expect(props.successEditFn).toHaveBeenCalledTimes(1);
        });
    });
});
