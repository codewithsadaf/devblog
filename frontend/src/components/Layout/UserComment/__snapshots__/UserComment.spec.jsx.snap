// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserComment /> Rendering should match snapshot 1`] = `
.c8 {
  font-size: 1em;
  line-height: 150%;
}

.c3 {
  width: 150px;
  height: 150px;
  object-fit: cover;
  display: block;
  margin: 1em auto;
  border-radius: 7px;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  margin: 2em 0;
  padding: 0 7px;
  position: relative;
}

.c1 {
  min-width: 7%;
  width: 7%;
  height: 100%;
  margin: 0 1em 0 0;
}

.c1 .c2 {
  width: 100%;
  height: 100%;
  margin: 0;
}

.c4 {
  width: 93%;
  border-bottom: 1px solid #ccc;
  padding-bottom: 1em;
}

.c4 > .c7 {
  margin-top: 1.2em;
  display: block;
  width: 100%;
  word-wrap: break-word;
}

.c5 {
  text-transform: none;
  font-weight: bold;
  font-size: 1.2em;
  line-height: 120%;
}

.c6 {
  text-transform: none;
  font-weight: 500;
  font-size: 0.8em;
  line-height: 150%;
}

@media (max-width:768px) {
  .c1 {
    width: 10%;
    min-width: 10%;
  }
}

@media (max-width:768px) {
  .c4 {
    width: 90%;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class=""
    >
      <img
        alt="User Image"
        class="c2 c3"
        src="https://example.com/user.jpg"
      />
    </div>
  </div>
  <div
    class="c4"
  >
    <h6
      class="c5"
    >
      John Doe
    </h6>
    <h1
      class="c6"
    >
      2023-04-29
    </h1>
    <p
      class="c7 c8"
    >
      Lorem ipsum dolor sit amet
    </p>
  </div>
</div>
`;
